### PR TITLE
Add go mod for E2E

### LINKED ISSUE #512: aws-iam-authenticator v0.6.0 breaks imports
We use `sigs.k8s.io/aws-iam-authenticator/pkg/token` to allow some of our dev tooling to use IAM credentials for authentication, which means we import `sigs.k8s.io/aws-iam-authenticator` as a module.

In past versions, this works well. Unfortunately, the inclusion of `k8s.io/kubernetes` in the go.mod for aws-iam-authenticator v0.6.0 means it's no longer straight-forward to import, because of all the [local path replacements](https://github.com/kubernetes/kubernetes/blob/v1.22.0/go.mod#L489-L520). This inclusion was introduced by #476, which is a great addition in itself.

For example, with v0.5.4, the following works just fine:

```
❯ mkdir foobar
❯ cd foobar
❯ git init
Initialized empty Git repository in /Users/ripta/projects/foobar/.git/
❯ go mod init github.com/ripta/foobar
go: creating new go.mod: module github.com/ripta/foobar

❯ go get sigs.k8s.io/aws-iam-authenticator@v0.5.4
go: added sigs.k8s.io/aws-iam-authenticator v0.5.4
```

but in v0.6.0, the `go get` step fails with:

```
❯ go get sigs.k8s.io/aws-iam-authenticator@v0.6.0
go: sigs.k8s.io/aws-iam-authenticator@v0.6.0 requires
	k8s.io/kubernetes@v1.22.0 requires
	k8s.io/api@v0.0.0: reading k8s.io/api/go.mod at revision v0.0.0: unknown revision v0.0.0
```

I'm trying to figure out a long-term solution. Right now, I think we'd be stuck having to replicate the [explicit replacements you do](https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/v0.6.0/go.mod#L28-L57) in our own go.mod.

I know we're [not the only ones](https://pkg.go.dev/sigs.k8s.io/aws-iam-authenticator/pkg/token?tab=importedby) that import this repo to get access to `pkg/token`.

Thanks!

### PR DESCRIPTION
**Pull request by nckturner**
**What this PR does / why we need it**:

* Add separate go mod for E2E
* Remove the kubernetes dependency from the main go.mod
* This allows other projects to import aws-iam-authenticator as a go module without including the same replace directives as kubernetes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #512, #514

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,57 +1,84 @@
 module sigs.k8s.io/aws-iam-authenticator
 
-go 1.16
```
Comment (nnmin-aws) [inline go.mod:3]: this defines lowest go version. will user have to upgrade from 1.16/1.17/1.18?

Comment (nckturner) [reply go.mod:3]: I think we should upgrade the version (at least to the last supported version by the go team, which is 1.18 right now I believe) but I'll do it in a separate PR

**Review by micahhausler — APPROVED** [review]
/lgtm
/approve

**Pull request merged**